Automattic/wp-calypso
# Domains: Remove UsersStore in favor of `react-query`

#### Changes proposed in this Pull Request

With #51784 landing, now we have a single use of the users store in the `TransferOtherUser` component - part of the flow to transfer a domain to another user. This PR removes that last instance and does some cleanup. Full list of the changes:

* Introduce a simple `withUser` higher-order component to allow us to use `react-query` without having to refactor a component to a functional one.
* Refactor `TransferOtherUser` to use `withUser` instead of the legacy users store.
* Cleanup `DomainManagementData` from any legacy users store handling, as it's no longer used anywhere.
* Cleanup from `StoreConnection` and remove it completely.

This PR is part of #24150 and the unified reduxification (de-fluxification) effort. 😉 

#### Testing instructions

* Verify `/domains/manage/:domain/transfer/other-user/:site` still works like it did before.
* Specifically, verify it correctly loads the users that are eligible for a domain transfer.

#### Notes

I'm pinging some teams for review:
* @Automattic/cobalt and @aidvu - because this touches Domains
* @Automattic/team-calypso-frameworks - because this PR removes more Flux!
